schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java

```diff
@@ -278,31 +278,52 @@ private SchemaVersionInfo createSchemaVersion(SchemaMetadata schemaMetadata,
                         if (schemaVersionInfos != null && !schemaVersionInfos.isEmpty()) {
                             SchemaVersionInfo latestSchemaVersionInfo = null;
                             SchemaCompatibility compatibility = schemaMetadata.getCompatibility();
+                            SchemaValidationLevel validation = schemaMetadata.getValidation();
                             for (SchemaVersionInfo schemaVersionInfo : schemaVersionInfos) {
+                                if (SchemaValidationLevel.ALL.equals(validation)) {
+                                    // check for compatibility
+                                    CompatibilityResult compatibilityResult =
```

Comment:
Below code can be extracted into a method and which can also be used for condition `SchemaValidationLevel.LATEST.equals(validation)`

```
    // check for compatibility
    CompatibilityResult compatibilityResult =
        schemaTypeWithProviders
            .get(type)
            .checkCompatibility(schemaText,
                                schemaVersionInfo.getSchemaText(),
                                compatibility);
    if (!compatibilityResult.isCompatible()) {
        String errMsg = String.format("Given schema is not compatible with earlier schema versions. \n" +
                                      "Error location: [%s] \n" +
                                      "Error encountered is: [%s]",
                                      compatibilityResult.getErrorLocation(),
                                      compatibilityResult.getErrorMessage());
        LOG.error(errMsg);
        throw new IncompatibleSchemaException(errMsg);
    }
```